src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
+     */
+    matchShape() {
+        return d => {
+            if (d.shape === undefined)  { return d3.symbolCircle   }
+            if (d.shape === "cross")    { return d3.symbolCross    }
+            if (d.shape === "diamond")  { return d3.symbolDiamond  }
+            if (d.shape === "square")   { return d3.symbolSquare   }
+            if (d.shape === "star")     { return d3.symbolStar     }
+            if (d.shape === "triangle") { return d3.symbolTriangle }
+            return d3.symbolCircle
+        };
+    }
+
+    /**
+     * Creates labels on axes if they're defined.
+     */
+    createLabels(axis,svg,box_width,margin,box_height) {
+        let fontStyle = "10px DejaVuSansMonoBook";
+        if (axis.x.label !== undefined) {
+            let padding_y = 20;
+            svg.append("text")
+                .attr("text-anchor","end")
+                .attr("style",label_style)
+                .attr("x",margin.left + (this.getTextWidth(axis.x.label,fontStyle) / 2))
+                .attr("y",box_height + margin.top + padding_y)
+                .text(axis.x.label);
+        }
+
+        if (axis.y.label !== undefined) {
+            let padding_y = 15;
+            svg.append("text")
+                .attr("text-anchor","end")
+                .attr("style",label_style)
+                .attr("transform","rotate(-90)")
+                .attr("y",-margin.left + padding_y)
+                .attr("x",-margin.top - (box_height/2) + (this.getTextWidth(axis.y.label,fontStyle) / 2))
+                .text(axis.y.label);
+        }
+    }
+
+    /**
+     * Helper function to get text width to make sure text wont overlap on screen.
+     */
+    getTextWidth(text,font) {
+        const canvas  = document.createElement("canvas");
+        const context = canvas.getContext("2d");
+        context.font  = font;
+        const metrics = context.measureText("  " + text);
+        return metrics.width;
+    }
+
+    /**
+     * Creates plot's axes.
+     */
+    createAxes(axis,extremesAndDeltas,box_width,box_height,svg,focus) {
+        let {domain_x,domain_y} = this.getDomains(extremesAndDeltas,focus);
+
+        let xScale = d3.scaleLinear();
+        if (axis.x.scale !== "linear") { xScale = d3.scaleLog(); }
+
+        xScale.domain(domain_x).range([0,box_width]);
+        let xAxis = svg.append("g")
+            .attr("transform","translate(0," + box_height + ")")
+            .attr("style",label_style)
+            .call(d3.axisBottom(xScale).ticks(box_width/x_axis_label_width))
+
+        let yScale = d3.scaleLinear()
+        if (axis.y.scale !== "linear") { yScale = d3.scaleLog(); }
+
+        yScale.domain(domain_y).range([box_height,0]);
+        let yAxis = svg.append("g")
+            .attr("style",label_style)
+            .call(d3.axisLeft(yScale));
+        return {xScale:xScale,yScale:yScale,xAxis:xAxis,yAxis:yAxis};
+    }
+
+    /**
+     * Helper function calculating domains of given data.
+     */
```

Comment:
this docs does not tell anything. What is "domain" what is "given data"? Need to be better descirbed, preferably with examples